### PR TITLE
Prometheus metrics for keeper

### PR DESCRIPTION
**Pull request by connorwstein**
Temporary until 6065

### CODE REVIEW

File: core/services/pipeline/runner.go [hunk 1]
```diff
@@ -48,12 +48,31 @@ type runner struct {
...
 var (
-	promPipelineTaskExecutionTime = promauto.NewGaugeVec(prometheus.GaugeOpts{
+	// TODO: Make private again after 6065
+	PromPipelineTaskExecutionTime = promauto.NewGaugeVec(prometheus.GaugeOpts{
 		Name: "pipeline_task_execution_time",
 		Help: "How long each pipeline task took to execute",
 	},
 		[]string{"job_id", "job_name", "task_type"},
 	)
+	PromPipelineRunErrors = promauto.NewCounterVec(prometheus.CounterOpts{
```
Comment (connorwstein) [inline core/services/pipeline/runner.go:58]: Also just grouping these in one place

File: core/services/keeper/upkeep_executer.go [hunk 2]
```diff
@@ -239,6 +239,19 @@ func (executer *UpkeepExecuter) execute(upkeep UpkeepRegistration, headNumber in
 	if err != nil {
 		logger.Errorw("UpkeepExecuter: failed to update database state", "err", err)
 	}
+
+	// TODO: Remove in 6065
```
Comment (samsondav) [inline core/services/keeper/upkeep_executer.go:243]: Can I get a link to the story?

File: core/services/chainlink/application.go [hunk 3]
```diff
@@ -677,6 +677,19 @@ func (app *ChainlinkApplication) RunJobV2(
 			CreatedAt:      t,
 			FinishedAt:     &t,
 		}, nil, false)
+		elapsed := time.Since(t)
```
Comment (samsondav) [inline core/services/chainlink/application.go:680]: Feels icky having all this logic in `application.go` - can it move into the runner (I mean, the meat of this entire function)?

File: core/services/pipeline/runner.go [hunk 4]
```diff
@@ -48,12 +48,31 @@ type runner struct {
 }
 
 var (
-	promPipelineTaskExecutionTime = promauto.NewGaugeVec(prometheus.GaugeOpts{
+	// TODO: Make private again after 6065
```
Comment (samsondav) [inline core/services/pipeline/runner.go:51]: Link pls

**Review by samsondav — DISMISSED** [review]

File: core/services/chainlink/application.go [hunk 5]
```diff
@@ -677,6 +677,19 @@ func (app *ChainlinkApplication) RunJobV2(
...
+
+		// For testing metrics.
+		pipeline.PromPipelineTaskExecutionTime.WithLabelValues(fmt.Sprintf("%d", jb.ID), jb.Name.String, jb.Type.String()).Set(float64(elapsed))
```
Comment (j16r) [inline core/services/chainlink/application.go:683]: Worth saving `id := fmt.Sprintf("%d", jb.ID)` ?

File: core/services/chainlink/application.go [hunk 6]
```diff
@@ -677,6 +677,19 @@ func (app *ChainlinkApplication) RunJobV2(
...
+		var status string
+		if err != nil {
+			status = "error"
+			pipeline.PromPipelineRunErrors.WithLabelValues(fmt.Sprintf("%d", jb.ID), jb.Name.String).Inc()
```
Comment (j16r) [inline core/services/chainlink/application.go:687]: What about just having a counter for completed/error and setting a label for status?

Comment (connorwstein) [reply core/services/chainlink/application.go:687]: This is just the same as the existing pattern which we've already shipped - I don't think its worth the churn for nops

**Review by samsondav — APPROVED** [review]

**Pull request merged**